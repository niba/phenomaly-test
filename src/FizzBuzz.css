.blue {
  color: blue;
}
.green {
  color: green;
}
.yellow {
  color: yellow;
}