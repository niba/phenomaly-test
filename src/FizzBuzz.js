import React, { Component } from 'react';
import Button from './Common/Button';
import './FizzBuzz.css';

class FizzBuzz extends Component {

  constructor() {
    super();

    this.state = {
      counter: 1
    };

    this.incrementCounter = this.incrementCounter.bind(this);
  }

  incrementCounter() {
    this.setState({
      counter: this.state.counter + 1
    });
  }

  checkNumber(number) {
    if (number % 5 === 0 && number % 3 === 0) {
      return <span className="green">{"FizzBuzz"}</span>;
    }
    if (number % 5 === 0) {
      return <span className="blue">{"Buzz"}</span>;
    }
    if (number % 3 === 0) {
      return <span className="yellow">{"Fizz"}</span>;
    }
    return number;
  }

  render() {
    return (
      <div>
         <span style={{ marginRight: '10px', fontWeight: 'bold' }}> {this.checkNumber(this.state.counter)} </span>
         <Button onClick={this.incrementCounter} text="increment" />
      </div>
    );
  }
}



export default FizzBuzz;
