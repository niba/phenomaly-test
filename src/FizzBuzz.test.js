import React from 'react';
import renderer from 'react-test-renderer';
import FizzBuzz from './FizzBuzz';

it('check fizz', () => {
  const component = renderer.create(
    <FizzBuzz />
  );
  component.getInstance().incrementCounter();
  component.getInstance().incrementCounter();
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

it('check buzz', () => {
  const component = renderer.create(
    <FizzBuzz />
  );

  for (let i = 1; i < 10; i++) {
     component.getInstance().incrementCounter();
  }

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

it('check fizz buzz', () => {
  const component = renderer.create(
    <FizzBuzz />
  );
  for (let i = 1; i < 30; i++) {
     component.getInstance().incrementCounter();
  }
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})