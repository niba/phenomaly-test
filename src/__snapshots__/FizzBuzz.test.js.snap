exports[`test check buzz 1`] = `
<div>
  <span
    style={
      Object {
        "fontWeight": "bold",
        "marginRight": "10px",
      }
    }>
     
    <span
      className="blue">
      Buzz
    </span>
     
  </span>
  <div
    className="btn"
    onClick={[Function]}
    style={undefined}>
    <span>
       
      increment
       
    </span>
  </div>
</div>
`;

exports[`test check fizz 1`] = `
<div>
  <span
    style={
      Object {
        "fontWeight": "bold",
        "marginRight": "10px",
      }
    }>
     
    <span
      className="yellow">
      Fizz
    </span>
     
  </span>
  <div
    className="btn"
    onClick={[Function]}
    style={undefined}>
    <span>
       
      increment
       
    </span>
  </div>
</div>
`;

exports[`test check fizz buzz 1`] = `
<div>
  <span
    style={
      Object {
        "fontWeight": "bold",
        "marginRight": "10px",
      }
    }>
     
    <span
      className="green">
      FizzBuzz
    </span>
     
  </span>
  <div
    className="btn"
    onClick={[Function]}
    style={undefined}>
    <span>
       
      increment
       
    </span>
  </div>
</div>
`;
